tests/functional/check_overstays_test.go

```diff
@@ -0,0 +1,95 @@
+package functional
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/IgorSteps/easypark/internal/adapters/rest/models"
+	"github.com/IgorSteps/easypark/internal/domain/entities"
+	"github.com/IgorSteps/easypark/tests/functional/client"
+	"github.com/IgorSteps/easypark/tests/functional/utils"
+	"github.com/stretchr/testify/suite"
+)
+
+type CheckOverStaysTestSuite struct {
+	client.RestClientSuite
+}
+
+func (s *CheckOverStaysTestSuite) TestCheckOverStays() {
+	// --------
+	// ASSEMBLE
+	// --------
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Create and login an admin
+	adminToken := utils.CreateAndLoginAdmin(ctx, &s.RestClientSuite)
+
+	// Create and login a driver
+	driver, driverToken := utils.CreateAndLoginDriver(ctx, &s.RestClientSuite, nil)
+
+	// Create a parking lot with a capacity of 5
+	createParkingLot := &models.CreateParkingLotRequest{
+		Name:     "test-lot",
+		Capacity: 5,
+	}
+	parkingLot := utils.CreateParkingLot(ctx, adminToken, createParkingLot, &s.RestClientSuite)
+	parkingSpaceID := parkingLot.ParkingSpaces[0].ID
+
+	// Create a parking request that will end in 3 seconds
+	createParkingRequest := &models.CreateParkingRequestRequest{
+		DestinationParkingLotID: parkingLot.ID,
+		StartTime:               time.Now(),
+		EndTime:                 time.Now().Add(3 * time.Second),
+	}
+	parkingRequest := utils.CreateParkingRequest(ctx, driverToken, driver.ID, parkingLot.ID, createParkingRequest, &s.RestClientSuite)
+	utils.AssignParkingSpace(ctx, parkingSpaceID, parkingRequest.ID, adminToken, &s.RestClientSuite)
+
+	// Create another parking request that will end in 2 hours
+	createParkingRequest2 := &models.CreateParkingRequestRequest{
+		DestinationParkingLotID: parkingLot.ID,
+		StartTime:               time.Now(),
+		EndTime:                 time.Now().Add(2 * time.Hour),
+	}
+	parkingRequest2 := utils.CreateParkingRequest(ctx, driverToken, driver.ID, parkingLot.ID, createParkingRequest2, &s.RestClientSuite)
+	utils.AssignParkingSpace(ctx, parkingSpaceID, parkingRequest2.ID, adminToken, &s.RestClientSuite)
+
+	// Send the arrival notification for the first parking request
+	testRequest := &models.CreateNotificationRequest{
+		ParkingRequestID: parkingRequest.ID,
+		ParkingSpaceID:   parkingSpaceID,
+		Location:         "parkingSpaceName",
```

Comment:
This must match with the actual parking space name. You can get it from the create parking lot response like you do the parking space id